subiquity/common/filesystem/manipulator.py

```diff
@@ -78,8 +78,11 @@ def delete_filesystem(self, fs):
     delete_format = delete_filesystem
 
     def create_partition(self, device, gap, spec, **kw):
+        flag = kw.pop('flag', None)
+        if gap.in_extended:
+            flag = 'logical'
```

Comment:
Should this yell at you if the passed flag is not None or logical?

Reply:
Yes, I think yelling about invalid flags is sensible.
But this could be any valid curtin partition flag I think?

Reply:
#1385